src/sentry/static/sentry/app/stores/organizationEnvironmentsStore.jsx

```diff
@@ -22,12 +34,26 @@ const OrganizationEnvironmentsStore = Reflux.createStore({
     };
   },
 
-  loadInitialData(environments) {
-    this.items = environments.map(this.makeEnvironment);
+  onFetchEnvironments() {
+    this.environments = null;
+    this.error = null;
+    this.trigger(this.get());
+  },
+
+  onFetchEnvironmentsSuccess(environments) {
```

Comment:
Let's error if `environments` is falsey

Reply:
sounds good, should we have the environments actionCreator contain this logic to check if the api returns a falsey value and dispatch an error action in that case or just leave the logic in the store?

Reply:
Yeah actionCreator sounds good - that way if multiple stores listen to this action they will all get the error action